.github/workflows/pr.yml

```
@@ -174,24 +187,35 @@ jobs:
         -configuration "Release" \
         | xcbeautify
 
-  asana:
+  create-asana-task:
     name: Create Asana Task
     needs: [swiftlint, unit-tests, shellcheck, release-build]
 
-    if: failure() && github.ref_name == 'develop'
-    
-    env:
-      WORKFLOW_URL: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
-    
+    if: failure() && github.ref_name == 'develop' && github.run_attempt == 1
+
     runs-on: ubuntu-latest
 
     steps:
       - name: Create Asana Task
-        uses: malmstein/github-asana-action@master
+        uses: duckduckgo/BrowserServicesKit/.github/actions/asana-failed-pr-checks@dominik/gha-pr-check-failed-improvements
```

After BSK is merged, this line needs to be rewritten as follows:
```suggestion
        uses: duckduckgo/BrowserServicesKit/.github/actions/asana-failed-pr-checks@main
```